staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt-config.go

```diff
@@ -0,0 +1,250 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	cache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	rmtypesv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+)
+
+// initialSync does the initial setup of the configuration of the
+// reqmgmt filter.  It either succeeds and returns `true` or logs an
+// error and returns `false.
+func (reqMgmt *requestManagement) initialSync() bool {
+	rms := &RMState{
+		flowSchemas:         make(FlowSchemaSeq, 0),
+		priorityLevelStates: make(map[string]*PriorityLevelState),
+	}
+	reqMgmt.curState.Store(rms)
+	reqMgmt.plInformer.AddEventHandler(reqMgmt)
+	reqMgmt.fsInformer.AddEventHandler(reqMgmt)
+	stopCh := make(chan struct{})
+	go reqMgmt.plInformer.Run(stopCh)
+	go reqMgmt.fsInformer.Run(stopCh)
+	if ok := cache.WaitForCacheSync(stopCh, reqMgmt.plInformer.HasSynced, reqMgmt.fsInformer.HasSynced); !ok {
+		return false
+	}
+	ok := reqMgmt.syncOne()
+	if !ok {
+		klog.Error("Unable to do initial config sync")
+		return false
+	}
+	go reqMgmt.Run(stopCh)
+	return true
+}
+
+// OnAdd handles notification from an informer of object creation
+func (reqMgmt *requestManagement) OnAdd(obj interface{}) {
+	reqMgmt.configQueue.Add(0)
+}
+
+// OnUpdate handles notification from an informer of object update
+func (reqMgmt *requestManagement) OnUpdate(oldObj, newObj interface{}) {
+	reqMgmt.OnAdd(newObj)
+}
+
+// OnDelete handles notification from an informer of object deletion
+func (reqMgmt *requestManagement) OnDelete(obj interface{}) {
+	reqMgmt.OnAdd(obj)
+}
+
+func (reqMgmt *requestManagement) Run(stopCh <-chan struct{}) {
+	defer reqMgmt.configQueue.ShutDown()
+	klog.Info("Starting reqmgmt config controller")
+	go wait.Until(reqMgmt.runWorker, time.Second, stopCh)
+	klog.Info("Started reqmgmt config worker")
+	<-stopCh
+	klog.Info("Shutting down reqmgmt config worker")
+}
+
+func (reqMgmt *requestManagement) runWorker() {
+	for reqMgmt.processNextWorkItem() {
+	}
+}
+
+func (reqMgmt *requestManagement) processNextWorkItem() bool {
+	obj, shutdown := reqMgmt.configQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	func(obj interface{}) {
+		defer reqMgmt.configQueue.Done(obj)
+		if !reqMgmt.syncOne() {
+			reqMgmt.configQueue.AddRateLimited(obj)
+		} else {
+			reqMgmt.configQueue.Forget(obj)
+		}
+	}(obj)
+
+	return true
+}
+
+// syncOne attempts to sync all the config for the reqmgmt filter.  It
+// either succeeds and returns `true` or logs an error and returns
+// `false`.
+func (reqMgmt *requestManagement) syncOne() bool {
+	all := labels.Everything()
+	newPLs, err := reqMgmt.plLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list PriorityLevelConfiguration objects: %s", err.Error())
+		return false
+	}
+	newFSs, err := reqMgmt.fsLister.List(all)
```

Comment:
It could definitely be an issue in my own code but in attempting to use the fsLister and a fakeClient, newFSs had nil values populated.  This also caused errors in the `func (a FlowSchemaSeq) Less(i, j int) bool {` method which does not handle nil values. Not sure if anything is needed, just thought I would mention issues I ran into.

Reply:
You mean the slice of `*FlowSchema` contained `nil`s?  That definitely should not happen!